### PR TITLE
build: update minimum required CMake to 3.13

### PR DESCRIPTION
**Pull request by spalicki**
Updating minimum required CMake to 3.13, also adding a fix for OpenMP (based on #2051) check on macOS since Apple's Clang compiler id changed to `AppleClang`.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -14,12 +14,17 @@
...
     cmake_policy(SET CMP0022 NEW)
 endif()
 
+# Compiler id for Apple Clang is now AppleClang
+if(POLICY CMP0025)
+    cmake_policy(SET CMP0025 NEW)
+endif()
+
 # Foo::Bar always refers to an IMPORTED target
 if(POLICY CMP0028)
```
Comment (vpirogov) [inline CMakeLists.txt:29]: (minor) Some of the policies might not be needed anymore with CMake version update.

File: cmake/OpenMP.cmake [hunk 2]
```diff
@@ -23,7 +23,7 @@ endif()
 set(OpenMP_cmake_included true)
 include("cmake/Threading.cmake")
 
-if (APPLE AND CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
+if (APPLE AND CMAKE_CXX_COMPILER_ID MATCHES "Clang")
```
Comment (densamoilov) [inline cmake/OpenMP.cmake:26]: It seems that `set_openmp_values_for_old_cmake` is no longer needed.

File: examples/CMakeLists.txt.in [hunk 3]
```diff
@@ -14,7 +14,7 @@
 # limitations under the License.
 #===============================================================================
 
-cmake_minimum_required(VERSION 2.8.12)
+cmake_minimum_required(VERSION 3.18.2)
```
Comment (densamoilov) [inline examples/CMakeLists.txt.in:17]: (random spot)
There are a few places in the build system that we want to adjust after this change:
* https://github.com/intel-innersource/libraries.performance.math.onednn/blob/e3c3d4799afe447f314eb00e4bcc1009833ab195/cmake/SYCL.cmake#L139
* https://github.com/intel-innersource/libraries.performance.math.onednn/blob/e3c3d4799afe447f314eb00e4bcc1009833ab195/cmake/platform.cmake#L316
* https://github.com/intel-innersource/libraries.performance.math.onednn/blob/e3c3d4799afe447f314eb00e4bcc1009833ab195/cmake/utils.cmake#L168
* https://github.com/intel-innersource/libraries.performance.math.onednn/blob/e3c3d4799afe447f314eb00e4bcc1009833ab195/examples/CMakeLists.txt.in#L93
* https://github.com/intel-innersource/libraries.performance.math.onednn/blob/e3c3d4799afe447f314eb00e4bcc1009833ab195/examples/CMakeLists.txt.in#L125

**Comment by spalicki** [comment]
make test

**Review by TaoLv — APPROVED** [review]

File: CMakeLists.txt [hunk 4]
```diff
@@ -14,7 +14,7 @@
 # limitations under the License.
 #===============================================================================
 
-cmake_minimum_required(VERSION 2.8.12)
+cmake_minimum_required(VERSION 3.18.2)
```
Comment (TaoLv) [inline CMakeLists.txt:17]: Just curious, how do we decide 3.18.2 as the minimum cmake version? For 2.8.12, I understand it's the default version distributed along with centos7. Thanks.

Comment (spalicki) [reply CMakeLists.txt:17]: It is the default version in RHEL8.

File: cmake/OpenMP.cmake [hunk 5]
```diff
@@ -23,38 +23,22 @@ endif()
...
```
Comment (avmanerikar) [inline cmake/OpenMP.cmake:26]: It will be useful to add a comment for this update and how it fixes the OpenMP check on MacOS in PR #2051.

**Review by vpirogov — APPROVED** [review]

**Comment by ilya-lavrenov** [comment]
> Updating minimum required CMake to 3.18.2

Why this version is selected? 
From the first glance, I don't see any cmake features used in oneDNN which require 3.18.2

**Comment by vpirogov** [comment]
@ilya-lavrenov, the intent here is to work out of the box with RHEL 8.5 and Ubuntu 20.04, which are the oldest distributions still supported by RHEL and Ubuntu. Do you have additional considerations?

**Comment by ilya-lavrenov** [comment]
> @ilya-lavrenov, the intent here is to work out of the box with RHEL 8.5 and Ubuntu 20.04, which are the oldest distributions still supported by RHEL and Ubuntu. Do you have additional considerations?

I prefer to use minimum required version based on cmake features used in the project.
In OpenVINO we use 3.13 as minimum version on Linux and having 3.18 in oneDNN will force us to the same requirements.

From what I see in oneDNN cmake files, 3.13 should also work for you. In future, you can iteratively increase version once new requirements are raised which forcing newer cmake versions to be used.

**Comment by vpirogov** [comment]
Thanks, @ilya-lavrenov. This sounds reasonable.

**Comment by spalicki** [comment]
@ilya-lavrenov @vpirogov Let me recompile using 3.13 as minimum and check if it works fine.

**Comment by spalicki** [comment]
make test

**Comment by spalicki** [comment]
@ilya-lavrenov @vpirogov I have changed the minimum version to 3.13.

**Review by ilya-lavrenov — APPROVED** [review]

**Review by hidefromkgb — APPROVED** [review]

File: cmake/SDL.cmake [hunk 6]
```diff
@@ -68,7 +68,7 @@ if(UNIX)
         sdl_gnu_common_ccxx_flags(ONEDNN_SDL_COMPILER_FLAGS CMAKE_CXX_COMPILER_VERSION)
         sdl_gnu_src_ccxx_flags(CMAKE_SRC_CCXX_FLAGS)
         sdl_gnu_example_ccxx_flags(CMAKE_EXAMPLE_CCXX_FLAGS)
-    elseif("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
+    elseif(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
```
Comment (densamoilov) [inline cmake/SDL.cmake:71]: I assume it's done to cover the AppleClang case. I think we should be more specific on what's allowed to show the intent. To do so we could use the following regex to cover Clang and AppleClang: `^(Clang|AppleClang)$` throughout the build system

By the way, it seems that there is one place in the build system where we already handle such a case:
https://github.com/oneapi-src/oneDNN/blob/c852fdc60073a1ae298618c4faf26487b4144755/cmake/coverage.cmake#L39
I think either regex will do.

**Review by densamoilov — APPROVED** [review]

**Comment by spalicki** [comment]
make test

**Review by vpirogov — APPROVED** [review]

**Pull request merged**